drivers: adt7420: update driver functions

ranechita opened the pull request:
The adt7420 had some reduntant code and some
hard to read functions. This commit aims to clear
the code and bring the driver up to date with
current no-os standards.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -92,38 +90,68 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+
+	regval &= ~mask;
+	regval |= value;
+	if (!adt7420_is_spi(dev) || register_address == ADT7320_REG_STATUS
```
This approach looks a bit odd. Since we are updating the driver, i'd suggest a more common approach (used also in the linux drivers) when registers have variable data size.

We should somehow store the data size of the registers in the macros:
https://github.com/analogdevicesinc/no-OS/blob/9f4b208d4b411b64707a8beed4cb0bac5dbbeace/drivers/dac/ad7293/ad7293.h#L56-L68

And process them accodingly in the spi read/write functions.

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -92,38 +90,68 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+		data_buffer[1] = regval & 0xFF;
+	}
+
+	if (adt7420_reg_write(dev, register_address, data_buffer))
```
nit: return directly

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -92,38 +90,68 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+*******************************************************************************/
+int adt7420_reg_write(struct adt7420_dev *dev,
+		      uint8_t register_address,
+		      uint8_t *data)
```
it's a bit uncommon to pass the data parameter as pointer in the write function. use a normal variable and based on the register size process the data bits.

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -92,38 +90,68 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
-			   uint8_t register_address,
-			   uint8_t num_data_bytes,
-			   uint8_t *data, uint8_t mask, uint8_t value)
+int adt7420_set_register_mask(struct adt7420_dev *dev,
```
nit: `adt7420_set_register_mask` -> `adt7420_reg_update_bits`

amiclaus replied:
this still needs to be addressed

amiclaus reviewed (commented):
some comments on my side.

adrimbarean reviewed (commented):
Looks good to me. I'd say we can merge after the comments have been addressed.

ranechita commented:
v2

- added an extra bit to the reg addresses to differentiate 16-bit registers
- returned status code of adt7420_reg_write directly
- removed unnecessary change
- used variable instead of pointer for reg_write

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -347,37 +366,29 @@ float adt7420_get_temperature(struct adt7420_dev *dev)
-		       uint8_t *data)
+ * @return register_value  - Value of the register.
+*******************************************************************************/
+uint16_t adt7420_spi_get_register_value(struct adt7420_dev *dev,
```
rename this also to `adt7420_reg_read` to be alligned with the rest for the functions, pass a pointer where to store the data, and return error code instead.

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -92,38 +90,63 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+	if(adt7420_is_spi(dev) && no_os_field_get(ADT7320_L16, register_address)) {
+		data_buffer[0] = regval >> 8;
+		data_buffer[1] = regval & 0xFF;
+	} else
```
nit: if we use brackets for the `if` statement, then the `else` should have also, even if it contains only one expression.

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -92,38 +90,63 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+						  register_address) << 3) & ADT7320_WRITE_MASK_CMD;
+		data_buffer[1] = (data >> 8) & 0xFF;
+		data_buffer[2] = data;
+		if (no_os_spi_write_and_read(dev->spi_desc, data_buffer, 3) != 0)
```
nit: `if (no_os_spi_write_and_read(dev->spi_desc, data_buffer, 3))`

amiclaus commented on drivers/temperature/adt7420/adt7420.c:
```diff
@@ -75,56 +73,84 @@ const struct adt7420_chip_info chip_info[] = {
 	uint8_t data_buffer[3] = {0, 0, 0};
+	uint16_t regval;
+
+	if(adt7420_reg_read(dev, register_address, &regval) != 0)
```
`if(adt7420_reg_read(dev, register_address, &regval))`

amiclaus replied:
btw, between, `if` and `(adt7420_reg_read(dev, register_address, &regval))` an empty space is required.

amiclaus reviewed (approved):
one more nit, otherwise lgtm. Also please try to provide a changelog after each force-push.
